packages/target-ethers-v5/src/codegen/index.ts

```diff
@@ -189,17 +189,9 @@ export function codegenContractFactory(contract: Contract, abi: any, bytecode?:
 
   const imports = createImportsForUsedIdentifiers(
     {
-      ethers: [
-        'Signer',
-        'utils',
-        'Contract',
-        'ContractFactory',
-        'PayableOverrides',
-        'Overrides',
-        'BytesLike',
-        'BigNumberish',
-      ],
-      '@ethersproject/providers': ['Provider', 'TransactionRequest'],
+      ethers: ['Signer', 'utils', 'Contract', 'ContractFactory', 'PayableOverrides', 'BytesLike', 'BigNumberish'],
```

Comment:
I can change this line to:
```suggestion
      'type ethers': ['Signer', 'utils', 'Contract', 'ContractFactory', 'PayableOverrides', 'BytesLike', 'BigNumberish'],
```
(merge with next line) and all tests still pass.

Reply:
Do you also want me to edit tests so when something that is a type is imported without `import type` or the other way around, we catch it in the tests?